BGL/include/CGAL/boost/graph/Euler_operations.h

```diff
@@ -585,6 +585,12 @@ add_face(const VertexRange& vr, Graph& g)
 
   std::vector<vertex_descriptor> vertices(vr.begin(), vr.end()); // quick and dirty copy
   unsigned int n = (unsigned int)vertices.size();
+  //check that every vertex is unique
+  std::sort(vertices.begin(), vertices.end());
+  if(std::unique(vertices.begin(), vertices.end()) != vertices.end())
+    return boost::graph_traits<Graph>::null_face();
+  vertices.clear();
+  vertices=std::vector<vertex_descriptor>(vr.begin(), vr.end());
```

Comment:
Your code does not avoid the allocation of another vector. What I suggested was:
```c++
vertices.clear();
vertices.insert(vr.begin(), vr.end());
```
But actually, we can avoid `clear()`, with just:
```c++
// re-fill the vector with the initial (unsorted) values
std::copy(vr.begin(), vr.end(), vertices.begin());
```

Reply:
And this should be benchmarked.  Note also  that most often it will be a triangle mesh.